File: 2425/MAISFII/T7/inicio/Usuario.dart
import 'dart:core';

class Usuario {
  String correo;
  String pass;
  String? nombre;
  int? telefono;

  Usuario(this.nombre, this.correo, this.pass);

  Usuario.telefono(this.nombre, this.correo, this.pass, this.telefono);

  Usuario.nombre(this.correo, this.pass, this.telefono);

  mostrarDatos() {
    print("Nombre ${nombre ?? "no indicado"}");
    print("Correo $correo");
    print("Pass $pass");
    print("Telefono ${telefono ?? "no definido"}");
  }
}

File: 2526/INSONLINE/TEMA1/inicio/src/Usuario.kt
open class Usuario(var nombre: String, var correo: String) {
    // constructores -> primario / secundarios
    // atributos
    // metodos
    var telefono: Int?=null
    constructor(nombre: String, correo: String, telefono:Int):this(nombre,correo){
        this.telefono = telefono;
    }

    open fun mostrarDatos(){
        println("nombre = ${nombre}")
        println("correo = ${correo}")
        println("telefono = ${telefono?:"no indicado"}")
    }
}